tests/integrations/test_pytorch_lightning.py

```diff
@@ -14,77 +14,242 @@
+        y_pred = y_hat.argmax(axis=1).cpu().detach().numpy()
+        return {"loss": loss,
+                "y_true": y_true,
+                "y_pred": y_pred}
 
-class TestPytorchLightning:
-    def test_learn(self):
-        # given (Subject)
-        neptune_logger = NeptuneLogger(
-            close_after_fit=False,
+    def validation_epoch_end(self, outputs):
+        for dl_idx in range(2):
+            y_true = np.array([])
+            y_pred = np.array([])
+            for results_dict in outputs[dl_idx]:
+                y_true = np.append(y_true, results_dict["y_true"])
+                y_pred = np.append(y_pred, results_dict["y_pred"])
+            acc = accuracy_score(y_true, y_pred)
+            self.log(f"val/loader_acc", acc)
+
+    def test_step(self, batch, batch_idx):
+        x, y = batch
+        y_hat = self(x)
+        loss = F.cross_entropy(y_hat, y)
+        self.log("test/loss", loss, prog_bar=False)
+        y_true = y.cpu().detach().numpy()
+        y_pred = y_hat.argmax(axis=1).cpu().detach().numpy()
+        for j in np.where(np.not_equal(y_true, y_pred))[0]:
+            img = np.squeeze(x[j].cpu().detach().numpy())
+            img[img < 0] = 0
+            img = img / np.amax(img)
+            self.neptune_logger.experiment["model_code/test/misclassified_images"].log(
+                neptune.types.File.as_image(img),
+                description="y_pred={}, y_true={}".format(y_pred[j], y_true[j]),
+            )
+        return {"loss": loss,
+                "y_true": y_true,
+                "y_pred": y_pred}
+
+    def test_epoch_end(self, outputs):
+        y_true = np.array([])
+        y_pred = np.array([])
+        for results_dict in outputs:
+            y_true = np.append(y_true, results_dict["y_true"])
+            y_pred = np.append(y_pred, results_dict["y_pred"])
+        acc = accuracy_score(y_true, y_pred)
+        self.log(f"test/acc", acc)
+
+
+class MNISTDataModule(pl.LightningDataModule):
+    def __init__(self, batch_size, normalization_vector):
+        super().__init__()
+        self.batch_size = batch_size
+        self.normalization_vector = normalization_vector
+        self.mnist_train = None
+        # self.mnist_val = None
+        self.mnist_val1 = None
+        self.mnist_val2 = None
+        self.mnist_test = None
+
+    def prepare_data(self):
+        MNIST(os.getcwd(), train=True, download=True)
+        MNIST(os.getcwd(), train=False, download=True)
+
+    def setup(self, stage):
+        # transforms
+        transform = transforms.Compose([transforms.ToTensor(),
+                                        transforms.Normalize(self.normalization_vector[0],
+                                                             self.normalization_vector[1])])
+        if stage == "fit":
+            mnist_train = MNIST(os.getcwd(), train=True, transform=transform)
+            self.mnist_train, self.mnist_val1, self.mnist_val2 = random_split(mnist_train, [50000, 5000, 5000])
```

Comment:
Maybe we would like to truncate the dataset as it will decrease time consumption.